### PR TITLE
Add dev mode error for fork

### PR DESCRIPTION
**Pull request by vansosnin**
| Q                        | A  |
| ------------------------ | ---  |
| Fixed Issues?            | #1941  |
| Patch: Bug Fix?          |  Probably :)  |
| Major: Breaking Change?  |  No  |
| Minor: New Feature?      |  No  |
| Tests Added + Pass?      | Yes |
| Any Dependency Changes?  |  No  |

Added error logging in dev mode if argument passed to `fork` is an effect.

### CODE REVIEW

**Comment by vansosnin** [comment]
Hmm, status lagged, but build completed

image

File: packages/core/types/index.d.ts [hunk 1]
```diff
@@ -241,8 +241,8 @@ export interface RunSagaOptions<A, S> {
 }
 
 export const CANCEL: string
-export const END: END
-export type END = END
+export const END: EndType
```
Comment (Andarist) [inline packages/core/types/index.d.ts]: was this change needed? 🤔

Comment (vansosnin) [reply packages/core/types/index.d.ts]: I'm not sure. But this is the only way I could fix [strange build error](https://travis-ci.org/redux-saga/redux-saga/jobs/592087832#L720).

**Comment by Andarist** [comment]
Thanks!
